js/stores/windowStore.js

```diff
@@ -750,6 +752,22 @@ const doAction = (action) => {
         }
         break
       }
+    case windowConstants.WINDOW_ON_TOGGLE_BOOKMARK:
+      {
+        // show the AddEditBookmarkHanger control; saving/deleting takes place there
+        const activeFrame = frameStateUtil.getActiveFrame(windowState)
+        let siteDetail = siteUtil.getDetailFromFrame(activeFrame, siteTags.BOOKMARK)
+        const key = siteUtil.getSiteKey(siteDetail)
+
+        if (key !== null) {
+          const site = appStoreRenderer.state.getIn(['sites', key], Immutable.Map())
+          siteDetail = siteDetail.set('parentFolderId', site.get('parentFolderId'))
+          siteDetail = siteDetail.set('customTitle', site.get('customTitle'))
+        }
+        siteDetail = siteDetail.set('location', UrlUtil.getLocationIfPDF(siteDetail.get('location')))
+        windowActions.setBookmarkDetail(siteDetail, siteDetail, null, action.isBookmarked, true)
```

Comment:
Yeah, we want to remove this, but sadly we need to keep it for now. You will see this pattern in other PR's as well. For this particular case we will remove it in #9710, because we will remove `setBookmarkDetail` as per our discussion.